relayer/client-tx.go

```diff
@@ -17,50 +18,88 @@ func (c *Chain) CreateClients(dst *Chain) (err error) {
 		return err
 	}
 
-	// Create client for the destination chain on the source chain if it doesn't exist
-	if srcCs, err := c.QueryClientState(srcH.Header.Height); err != nil && srcCs == nil {
+	// Create client for the destination chain on the source chain if client id is unspecified
+	if c.PathEnd.ClientID == "" {
 		if c.debug {
 			c.logCreateClient(dst, dstH.Header.Height)
 		}
 		ubdPeriod, err := dst.QueryUnbondingPeriod()
 		if err != nil {
 			return err
 		}
-		consensusParams, err := dst.QueryConsensusParams()
+		msgs := []sdk.Msg{
+			c.PathEnd.CreateClient(
+				dstH,
+				dst.GetTrustingPeriod(),
+				ubdPeriod,
+				c.MustGetAddress(),
+			),
+		}
+		res, success, err := c.SendMsgs(msgs)
 		if err != nil {
 			return err
 		}
-		msg := c.PathEnd.CreateClient(
-			dstH,
-			dst.GetTrustingPeriod(),
-			ubdPeriod,
-			consensusParams,
-			c.MustGetAddress(),
-		)
-		clients.Src = append(clients.Src, msg)
+		if !success {
+			return fmt.Errorf("tx failed: %s", res.RawLog)
+		}
+
+		// update the client identifier
+		clientEvents := res.Logs[0].Events
+		for _, ev := range clientEvents {
+			if ev.Type == clienttypes.EventTypeCreateClient {
+				for _, attr := range ev.Attributes {
+					if attr.Key == clienttypes.AttributeKeyClientID {
+						c.PathEnd.ClientID = attr.Value
```

Comment:
this isn't that bad for state based event parsing

Reply:
Its not that bad, but its not that great.